ledger-service/http-json/src/main/scala/com/digitalasset/http/Endpoints.scala

```diff
@@ -118,73 +119,102 @@ class Endpoints(
     }
   }
 
+  def withJwtPayloadLoggingContext[A](jwtPayload: JwtPayload)(
+      fn: LoggingContextOf[JwtPayload with InstanceUUID with RequestID] => A
+  )(implicit lc: LoggingContextOf[InstanceUUID with RequestID]): A =
+    withEnrichedLoggingContext(
+      LoggingContextOf.label[JwtPayload],
+      Map(
+        "ledger_id" -> jwtPayload.ledgerId.toString,
+        "act_as" -> jwtPayload.actAs.toString,
+        "application_id" -> jwtPayload.applicationId.toString,
+        "read_as" -> jwtPayload.readAs.toString,
+      ),
+    ).run(fn)
+
+  // I think this is not the best name.
+  // Naming things is hard :/
+  def handleCommand[T[_]](req: HttpRequest)(
```

Comment:
Seems like there is a fair bit of duplication between the write & read endpoints. Can we share this? In the end both tokens can provide us with those fields so we should be able to make them extend a common trait (maybe we even have one already I forgot).

Reply:
I thought about union both jwtpayloads with a trait, however because it hasn't been done in the past already I hesitated to do it as both are similar but used in different scenarios.

However I would be much happier too with adding a trait and removing that duplicate code.

Reply:
adressed with 0b0c824 & 24b0c33